api/redhatopenshift/HcpCluster.Management/hcpCluster-models.tsp

```diff
@@ -491,3 +510,297 @@ model HcpOpenShiftClusterAdminCredential {
  *   End HCP cluster credentials
  * =======================================
  */
+
+/*
+ * =======================================
+ *  ExternalAuth resources
+ * =======================================
+ */
+
+/** External Auth profile */
+model ExternalAuthProperties {
+  /** Provisioning state */
+  @visibility(Lifecycle.Read)
+  provisioningState?: ExternalAuthProvisioningState;
+
+  /** An observation of the current state with additional information. */
+  @visibility(Lifecycle.Read)
+  condition?: ExternalAuthCondition;
+
+  /** Token Issuer profile */
+  issuer: TokenIssuerProfile;
+
+  /** External Auth OIDC clients
+   * There must not be more than 20 entries and entries must have unique namespace/name pairs.
+   */
+  @maxItems(20)
+  @OpenAPI.extension(
+    "x-ms-identifiers",
+    #["component", "clientId", "extraScopes"]
+  )
+  clients?: ExternalAuthClientProfile[];
+
+  /** External Auth claim
+   * This configures how claims are validated and applied.
+   */
+  claim: ExternalAuthClaimProfile;
+}
+
+/** Condition defines an observation of the external auth state. */
+model ExternalAuthCondition {
```

Comment:
All the fields in `ExternalAuthCondition` should have `@visibility(Lifecycle.Read)`.  I think I may have mislead you about this elsewhere (I already forget where).  Also, being as these are all read-only, the `@minLength` and `@maxLength` decorators don't seem necessary.